### DISCUSSION TITLE
Load build YAML to Cue inside Golang code

### DISCUSSION THREAD

**Discussion by roylisto** [cue-lang/cue#1137] [Q&A] [answered]
Hi I have a YAML file
```
a:
  foo: bar

```
And try to load it to golang code

```
ctx := cuecontext.New()
yamlContent, _ := ioutil.ReadFile("test.yaml")
yamlValue := ctx.CompileBytes(yamlContent)
fmt.Println(yamlValue)

```

then I get this error
`_|_ // 2:8: reference "bar" not found`

but if I use
`foo: "bar"`
It works

is this a bug or expected by ctx.CompileBytes() ?

the context is I need to validate YAML files by cue definition inside golang code

**Comment by myitcv** [comment, marked as answer]
@roylisto the `cue.Context` methods are for loading CUE. You are looking for https://pkg.go.dev/cuelang.org/go@v0.4.0/encoding/yaml#Extract, the result of which can then be built using `cue.Context.BuildFile()`.

**Reply by roylisto** [reply]
oh okay, and I found all I need is this function https://pkg.go.dev/cuelang.org/go/pkg/encoding/yaml#Validate

**Reply by roylisto** [reply]
`yamlFile, _ := ioutil.ReadFile("test.yaml")

valid, err := yaml.Validate(yamlFile, cueValue)`

**Reply by myitcv** [reply]
Ah for plain yaml validation, yes.
